### DISCUSSION TITLE
Use both passwords and external IDPs within the same organization

### DISCUSSION THREAD

**Discussion by DavideWalder** [zitadel/zitadel#5320] [Ideas]
Use case: an organization wants to be able to use an external IDP for some of the users, but still be able to use username+password for some of the users.

Currently this results in two problems:
- MFA requirements are on the organization level, but users using an external IDP should be exempted
- Users using the external IDP are no longer automatically redirected to the IDP (since username and password is allowed), they have to select it manually (could be confusing, since SSO is mostly about seamless integration).

A possible solution is to use multiple organizations for the same customer, but increases complexity. Is there any other solution?

**Comment by fforootd** [comment]
Hm that's a good question.

I think we should pick this up for https://github.com/zitadel/zitadel/issues/5101 since we could then create an action run during login to verify if the user was external and if so skip MFA and if the user was internal we could prompt them.

@hifabienne any thoughts from you?

**Reply by hifabienne** [reply]
I think some of the problems will be covered with the idp tempaltes. Or at least it is planed (not sure if it is in the first version). We want to be able to configure on the idp what should be taken over to ZITADEL. eg. if the mfa is checked in the provider it should not be checked again in ZITADEL.

**Comment by u1040867** [comment]
Hi everyone, are there any updates on this issue?

**Comment by hifabienne** [comment]
Hei we have added it as a criterium to the following issue: https://github.com/zitadel/zitadel/issues/5654
"If allow register on the idp is enabled, the general allow registration from the login policy should be ignored"
